internal/cephfs/fsjournal.go

```diff
@@ -70,15 +81,44 @@ func checkVolExists(ctx context.Context, volOptions *volumeOptions, secret map[s
 	imageUUID := imageData.ImageUUID
 	vid.FsSubvolName = imageData.ImageAttributes.ImageName
 
-	_, err = getVolumeRootPathCeph(ctx, volOptions, cr, volumeID(vid.FsSubvolName))
-	if err != nil {
-		var evnf ErrVolumeNotFound
-		if errors.As(err, &evnf) {
+	var evnf ErrVolumeNotFound
+
+	if sID != nil || pvID != nil {
+		clone, cloneInfoErr := getcloneInfo(ctx, volOptions, cr, volumeID(vid.FsSubvolName))
+		if cloneInfoErr != nil {
+			if errors.As(cloneInfoErr, &evnf) {
+				err = j.UndoReservation(ctx, volOptions.MetadataPool,
```

Comment:
If we did not find the clone, we may still have left behind the snapshot (and possibly protected). We should call `cleanupCloneFromSubvolumeSnapshot` prior to `UndoReservation` if this is a `pvID` case.